app/scripts/lib/offscreen-bridge/ledger-offscreen-bridge.ts

```diff
@@ -147,9 +148,31 @@ export class LedgerOffscreenBridge
           if (response?.success) {
             resolve(response.payload || response.success);
           } else {
-            reject(
-              new Error(response?.payload?.error || 'Unknown Ledger error'),
-            );
+            // need to process the payload to get the error
+            // and then reject with the error
+            const error = response?.payload?.error;
+
+            if (
+              error &&
+              typeof error.statusCode === 'number' &&
+              error.statusCode > 0
+            ) {
+              // this is TransportStatusError, convert the SerializedLedgerError to a TransportStatusError
```

Comment:
Very nit: Same here.

```suggestion
              // This is TransportStatusError, convert the SerializedLedgerError to a TransportStatusError
```